### PR TITLE
Document Utility Classes w/ Demos!

### PR DESCRIPTION
**Pull request by rockymountainhigh1943**
## Jira

http://vjira2:8080/browse/BDS-462

## Summary

Adds documentation and demo within thebPattern Lab "utilities" section for all bolt utility classes (does not include @breakpoint rules).

## Details

This PR includes all of the work needed to create new pages within the Pattern Lab "utilities" section for each of the bolt utility classes. Demos and demo scss was written to enhance and make possible the demos. This PR does not include any of the "@breakpoint" examples (or details).

## How to test

1. Pull down this branch, run all the usual commands after pulling a new branch
1. Navigate to Pattern Lab "utilities" section and review each new page for accuracy in what it describes for implementation

### CODE REVIEW

File: apps/pattern-lab/src/styles/_docs.scss [hunk 1]
```diff
@@ -402,3 +402,288 @@ pre {
...
+    margin: .5rem;
+    position: relative;
+    width: calc(33% - 1rem);
+    vertical-align: top;
+    padding: 0.7rem;
+
+    code {
+      font-size: 0.7rem;
+    }
+    &[class*="hoverable"]:hover {
```
Comment (mikemai2awesome) [inline apps/pattern-lab/src/styles/_docs.scss]: Single quotes in CSS.

File: apps/pattern-lab/src/styles/_docs.scss [hunk 2]
```diff
@@ -402,3 +402,288 @@ pre {
...
+  @extend %bolt-demo-item-border;
+  h4+& {
+    @include bolt-margin-top(0.75rem);
+  }
+}
+
+.c-bolt-docs__visually-hidden-demo {
+  @extend %bolt-demo-wrapper;
+
+  input[type=checkbox]{
```
Comment (mikemai2awesome) [inline apps/pattern-lab/src/styles/_docs.scss]: Single quote here also, for consistency.

File: packages/global/styles/07-utilities/_utilities-text-align.scss [hunk 3]
```diff
@@ -19,3 +19,6 @@
 .u-bolt-text-align-justify {
   text-align: justify !important;
 }
+
+$bolt-alignment-props: ("right", "left", "center", "justify");
```
Comment (mikemai2awesome) [inline packages/global/styles/07-utilities/_utilities-text-align.scss]: Can single quotes be used here?

File: packages/global/styles/07-utilities/_utilities-text-decoration.scss [hunk 4]
```diff
@@ -13,3 +13,6 @@
 .u-bolt-text-decoration-line-through {
   text-decoration: line-through !important;
 }
+
+$bolt-text-decoration-props: ("none", "underline", "line-through");
```
Comment (mikemai2awesome) [inline packages/global/styles/07-utilities/_utilities-text-decoration.scss]: Can single quotes be used here?

File: apps/pattern-lab/src/styles/_docs.scss [hunk 5]
```diff
@@ -402,3 +402,288 @@ pre {
...
+    &:last-of-type {
+      @include bolt-margin-bottom(0);
+      border-bottom: 0;
+      padding-bottom: 0;
+    }
+  }
+  &__color {
+    @include bolt-margin-bottom(2rem);
+
+    .o-bolt-wrapper {
```
Comment (sghoweri) [inline apps/pattern-lab/src/styles/_docs.scss]: ???

**Comment by sghoweri** [comment]
@rockymountainhigh1943 @remydenton @mikemai2awesome @margoromo there’s still a bunch of things I’m seeing in the 
https://github.com/bolt-design-system/bolt/blob/0e1dd08d1102c62bd267c59074bbb919607b5a26/apps/pattern-lab/src/styles/_docs.scss file I really think we should clean up before I’m good to merge this in — what I mentioned in my original review comments for this.

In a nutshell, “demo-specific” CSS we’re writing should be sticking to the same general coding methodologies and standards we’re shooting for across the board.

I’ll fill in the details once I get in but I think it ultimately comes down to two things:

1. Are our out of the box styles we have for layout and typography not good enough to provide the styling polish for our own docs pages in PL without having to write a ton of new CSS in the process? 

If they aren’t, where are we lacking so those actual components / patterns / layouts we shipping can get tightened up?

2. Assuming there’s some truly unique “special snowflake” styles that aren’t satisfied by existing components, layouts and utility classes in the design system and — those special snowflake styles can’t broken down into reusable styles we ship for some specific reason (which we should call out) — nonetheless we should still be componentizing whatever else is left in our own codebase.

- BEM classes shouldn’t be content or purpose specific — they should be as agnostic as possible to allow us the most reuse possible

- Assuming the layout / spacing of things can’t be satisfied by utility classes, make sure we’re using the actual margin and padding utility classes provided by the design system, using the spacing scale provided by the system (ie small, medium, large, etc sizes). If there’s a reason why something we’re shipping isn’t good enough, this is when we’d flag it!

- Why didn’t you break out the demo-specific styles written into heir own standalone Sass partial(s)?

- What was the thought process / reasoning behind how certain demo sections get a particular amount of spacing? 

There’s already been a ton of work we’ve done building out and improving our Bands, Wrappers, Headlines / Text, Grid, Utility Classes, etc etc so better understanding how you make the call when to reuse vs build would be super helpful to make sure we make the design system itself / usage of the design system as slick and easy as possible.

**Pull request merged**